Extension/DefaultOptionsExtension.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Knp\Bundle\MenuBundle\Extension;
+
+use Knp\Menu\Factory\ExtensionInterface;
+use Knp\Menu\ItemInterface;
+
+/**
+ * An extension to configure the default menu options.
+ *
+ * @author Wouter J <wouter@wouterj.nl>
+ */
+class DefaultOptionsExtension implements ExtensionInterface
+{
+    /**
+     * @var array
+     */
+    private $options;
+
+    public function __construct(array $options)
+    {
+        $this->options = $options;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildOptions(array $options = array())
+    {
+        return array_merge($this->options, $options);
```

Comment:
I don't know what to use here. ``array_merge_recursive`` has a very weird behavior when associate arrays are involved, so it is not the solution either (it would be even worse). ``array_replace_recursive`` would already be a bit better as we mostly have associative arrays (but this one leads to an unexpected behavior when lists are merged together though, and this happens in extras, for instance for the list of matched routes)